packages/synthetix-main/contracts/storage/Market.sol

```diff
@@ -406,84 +426,137 @@ library Market {
      * NOTE: this function should be called before any pool alters its liquidity allocation (see `rebalance` above)
      */
     function distributeDebt(Data storage self, uint maxIter) internal {
-        // No debt to distribute if there is no liquidity.
-        if (self.debtDist.totalShares == 0) {
-            return;
-        }
-
         // Get the current and last distributed market balances.
         // Note: The last distributed balance will be cached within this function's execution.
         int256 targetBalance = totalBalance(self);
-        int256 distributedBalance = self.lastDistributedMarketBalance;
-        int256 outstandingBalance = targetBalance - distributedBalance;
+        int256 outstandingBalance = targetBalance - self.lastDistributedMarketBalance;
+
+        (, bool exhaustedUp) = bumpPoolsOut(self, outstandingBalance, maxIter);
+        (, bool exhaustedDown) = bumpPoolsIn(self, outstandingBalance, maxIter);
+
+        if (!exhaustedDown && !exhaustedUp && self.debtDist.totalShares > 0) {
+            self.debtDist.distributeValue(targetBalance - self.lastDistributedMarketBalance);
```

Comment:
Can use `outstandingBalance` here?

Reply:
can't because `self.lastDistributedMarketBalance` may have changed after calling above methods

yes, I know the function is semantically overloaded, or has a side effect of sorts, but using the returned value `actuallyDistributed` is actually harder to validate from a code verification/clarity standpoint than just reading `self.lastDistributedMarketBalance` directly

added a comment to clarify this